src/sql-parser/src/parser.rs

```diff
@@ -236,6 +239,25 @@ impl fmt::Display for ParserStatementError {
     }
 }
 
+impl From<LexerError> for ParserError {
+    fn from(value: LexerError) -> Self {
+        ParserError {
+            message: value.message,
+            pos: value.pos,
+        }
+    }
+}
+
+trait IntoIdent {
```

Comment:
Defining a trait for this method is a little funny since it's only used in this file and it's not like it's super important to have this be a method on `Keyword`. I'd instead make a free function like `fn ident_to_keyword(kw: Keyword) -> Ident`, or `impl From<Keyword> for Ident` and then you can just call `kw.into()` where you need to.

Reply:
I had initially gone with the `From<>` impl, but backed it out as we do chain off `into_ident()` in a few places and I'd need to use `as` so I opted for this method to keep the callsites clean. I'll swap this to a free function.